dipy/reconst/ivim.py

```diff
@@ -0,0 +1,391 @@
+            Default : 1e-7
+            Tolerance for convergence of minimization.
+
+        options : dict, optional
+            Dictionary containing gtol, ftol, eps and maxiter. This is passed
+            to leastsq. By default these values are set to
+            options={'gtol': 1e-7, 'ftol': 1e-7, 'eps': 1e-7, 'maxiter': 1000}
+
+        References
+        ----------
+        .. [1] Le Bihan, Denis, et al. "Separation of diffusion
+                   and perfusion in intravoxel incoherent motion MR
+                   imaging." Radiology 168.2 (1988): 497-505.
+        .. [2] Federau, Christian, et al. "Quantitative measurement
+                   of brain perfusion with intravoxel incoherent motion
+                   MR imaging." Radiology 265.3 (2012): 874-881.
+        """
+        ReconstModel.__init__(self, gtab)
+        self.split_b = split_b
+        self.bounds = bounds
+        self.tol = tol
+        self.options = options
+
+        if SCIPY_LESS_0_17 and self.bounds is not None:
+            e_s = "Scipy versions less than 0.17 do not support "
+            e_s += "bounds. Please update to Scipy 0.17 to use bounds"
+            raise ValueError(e_s)
+        else:
+            self.bounds = (np.array([0., 0., 0., 0.]),
+                           np.array([np.inf, 1., 0.1, 0.1]))
+
+    @multi_voxel_fit
+    def fit(self, data, mask=None):
+        """ Fit method of the Ivim model class
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel. A multi voxel decorator
+            will be applied to this fit method to scale it and apply it
+            to multiple voxels.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[:-1]
+
+        Returns
+        -------
+        IvimFit object
+
+        References
+        ----------
+        .. [1] Federau, Christian, et al. "Quantitative measurement
+                   of brain perfusion with intravoxel incoherent motion
+                   MR imaging." Radiology 265.3 (2012): 874-881.
+        """
+        # Call the function _estimate_S0_D to get initial x0 guess.
+        x0 = self.estimate_x0(data)
+        # Use leastsq to get ivim_params
+        params_in_mask = self._leastsq(data, x0)
+        return IvimFit(self, params_in_mask)
+
+    def predict(self, ivim_params):
+        """
+        Predict a signal for this IvimModel class instance given parameters.
+
+        Parameters
+        ----------
+        ivim_params : array
+            The ivim parameters as an array [S0, f, D_star and D]
+
+        Returns
+        -------
+        ivim_signal : array
+            The predicted IVIM signal using given parameters.
+        """
+        return ivim_function(ivim_params, self.gtab.bvals)
+
+    def estimate_x0(self, data):
+        """
+        Fit the ivim params using a two stage fit.
```

Comment:
Is the two stage fit discussed somewhere? I know it's being used but maybe this wording is left over from before because I haven't run across it yet. Maybe this is the place to discuss the details of in and define it for the user?